File: app/src/main/java/fr/paug/androidmakers/model/Ribbon.java
package fr.paug.androidmakers.model;

import android.support.annotation.DrawableRes;
import android.text.TextUtils;

import fr.paug.androidmakers.R;

public class Ribbon {

    public enum RibbonType {
        NONE, GDE, GDG; // TODO GOO, WT

        @DrawableRes
        public int getRibbonIcon() {
            switch (this) {
                case GDE:
                    return R.drawable.gde_logo;
                case GDG:
                    return R.drawable.gdg_logo;
                default:
                    return R.drawable.transparent;
            }
        }

        @DrawableRes
        public int getBadgeIcon() {
            switch (this) {
                case GDE:
                    return R.drawable.gde_badge;
                case GDG:
                    return R.drawable.gdg_badge;
                default:
                    return R.drawable.transparent;
            }
        }
    }

    public final RibbonType ribbonType;
    public final String title;
    public final String link;

    public Ribbon(String ribbonName, String title, String link) {
        this.ribbonType = getRibbonType(ribbonName);
        this.title = title;
        this.link = link;
    }

    private static RibbonType getRibbonType(String ribbonName) {
        if (!TextUtils.isEmpty(ribbonName)) {
            if (ribbonName.equalsIgnoreCase("GDE")) {
                return RibbonType.GDE;
            } else if (ribbonName.equalsIgnoreCase("GDG")) {
                return RibbonType.GDG;
            }
        }
        return RibbonType.NONE;
    }

}

File: app/src/main/java/fr/paug/androidmakers/model/Ribbon.kt
package fr.paug.androidmakers.model

import android.text.TextUtils
import androidx.annotation.DrawableRes
import fr.paug.androidmakers.R

class Ribbon(ribbonName: String?, title: String?, link: String?) {
    enum class RibbonType {
        NONE, GDE, GDG;

        // TODO GOO, WT
        @get:DrawableRes
        val ribbonIcon: Int
            get() = when (this) {
                GDE -> R.drawable.gde_logo
                GDG -> R.drawable.gdg_logo
                else -> R.drawable.transparent
            }

        @get:DrawableRes
        val badgeIcon: Int
            get() = when (this) {
                GDE -> R.drawable.gde_badge
                GDG -> R.drawable.gdg_badge
                else -> R.drawable.transparent
            }
    }

    val ribbonType: RibbonType?
    val title: String?
    val link: String?

    companion object {
        private fun getRibbonType(ribbonName: String?): RibbonType {
            if (!TextUtils.isEmpty(ribbonName)) {
                if (ribbonName.equals("GDE", ignoreCase = true)) {
                    return RibbonType.GDE
                } else if (ribbonName.equals("GDG", ignoreCase = true)) {
                    return RibbonType.GDG
                }
            }
            return RibbonType.NONE
        }
    }

    init {
        ribbonType = getRibbonType(ribbonName)
        this.title = title
        this.link = link
    }
}